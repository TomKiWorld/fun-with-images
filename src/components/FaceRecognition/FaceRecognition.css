.image-holder {
  width: 80%;
  max-width: 22rem;
  position: relative;
}

.image-holder .bounding-box {
  position: absolute;
  box-shadow: 0 0 0 0.1rem #0239B7 inset;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  cursor: pointer;
}

.image-holder .view-colors {
  position: absolute;
  right: 0.5rem;
  top: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #5CDADA;
  border-radius: 50%;
  width: 3rem;
  height: 3rem;
  border: 0.1rem solid white;
  z-index: 5;
}

.image-holder img {
  width:100%;
}
