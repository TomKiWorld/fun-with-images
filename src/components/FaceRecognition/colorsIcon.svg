<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 270.53 247.76"><defs><linearGradient id="a" y1="144.72" x2="270.53" y2="144.72" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#242728"/><stop offset="1" stop-color="#969696"/></linearGradient><linearGradient id="b" x1="66.02" y1="80.16" x2="97.08" y2="80.16" xlink:href="#a"/><linearGradient id="c" x1="34.2" y1="101.74" x2="65.27" y2="101.74" xlink:href="#a"/><linearGradient id="d" x1="29.74" y1="140.63" x2="60.81" y2="140.63" xlink:href="#a"/><linearGradient id="e" x1="56.72" y1="197.06" x2="101.96" y2="197.06" xlink:href="#a"/><linearGradient id="f" x1="132.07" y1="213.24" x2="177.31" y2="213.24" xlink:href="#a"/><linearGradient id="g" x1="202.95" y1="167.1" x2="248.2" y2="167.1" xlink:href="#a"/><linearGradient id="h" x1="193.84" y1="96.78" x2="239.08" y2="96.78" xlink:href="#a"/><linearGradient id="i" x1="139.33" y1="175.31" x2="221.95" y2="175.31" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#969696"/><stop offset="1" stop-color="#242728"/></linearGradient><linearGradient id="j" x1="87.73" y1="39.64" x2="143.44" y2="39.64" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#ccc"/><stop offset="1" stop-color="#969696"/></linearGradient><linearGradient id="k" x1="96.82" y1="45.05" x2="115.08" y2="45.05" xlink:href="#a"/><linearGradient id="l" x1="104.48" y1="38.63" x2="138.25" y2="38.63" xlink:href="#a"/><linearGradient id="m" x1="115.5" y1="51.44" x2="130.06" y2="51.44" xlink:href="#a"/><linearGradient id="n" x1="95.69" y1="36.58" x2="115.79" y2="36.58" xlink:href="#a"/></defs><g data-name="Layer 2"><g data-name="Paint color"><path d="M269 138.35c0 59-59.89 106.79-133.76 106.79S1.5 197.33 1.5 138.35c0-35.16 20.82-66.35 53.29-85.81 22-13.18 50.08 4.69 79.92 4.69 25.37 0 48.5-20 68.91-10.24C242.65 65.71 269 99.62 269 138.35z" fill="none" stroke-miterlimit="10" stroke-width="3" stroke="url(#a)"/><path d="M66 87.24A18.56 18.56 0 0077.4 91c9.44 0 17.1-6.55 17.1-14.63a12.9 12.9 0 00-3.77-9.17 14.23 14.23 0 016.35 11.41c0 8.07-7.65 14.62-17.1 14.62A20.63 20.63 0 0166 87.24z" fill="url(#b)"/><ellipse cx="77.4" cy="76.32" rx="17.11" ry="14.62" fill="#ff24f6"/><path d="M34.2 108.82a18.56 18.56 0 0011.38 3.71c9.45 0 17.11-6.55 17.11-14.63a12.91 12.91 0 00-3.78-9.17 14.21 14.21 0 016.36 11.41c0 8.07-7.66 14.62-17.11 14.62a20.67 20.67 0 01-13.96-5.94z" fill="url(#c)"/><ellipse cx="45.58" cy="97.9" rx="17.11" ry="14.62" fill="#8e0098"/><path d="M29.74 147.7a18.5 18.5 0 0011.38 3.71c9.45 0 17.1-6.55 17.1-14.62a12.93 12.93 0 00-3.77-9.18A14.25 14.25 0 0160.81 139c0 8.08-7.66 14.62-17.11 14.62a20.65 20.65 0 01-13.96-5.92z" fill="url(#d)"/><ellipse cx="41.12" cy="136.79" rx="17.11" ry="14.62" fill="#61efff"/><path d="M56.72 207.72a26.41 26.41 0 0016.57 5.59c13.76 0 24.92-9.87 24.92-22a19.87 19.87 0 00-5.5-13.83c5.53 4 9.25 10.35 9.25 17.2 0 12.17-11.15 22-24.91 22a29.64 29.64 0 01-20.33-8.96z" fill="url(#e)"/><ellipse cx="73.29" cy="191.27" rx="24.91" ry="22.04" fill="#0600ea"/><path d="M132.07 223.9a26.37 26.37 0 0016.57 5.6c13.76 0 24.91-9.87 24.91-22a19.84 19.84 0 00-5.49-13.83c5.53 4 9.25 10.35 9.25 17.19 0 12.17-11.15 22-24.91 22a29.61 29.61 0 01-20.33-8.96z" fill="url(#f)"/><ellipse cx="148.64" cy="207.46" rx="24.91" ry="22.04" fill="#3cff5c"/><path d="M203 177.76a26.42 26.42 0 0016.58 5.6c13.75 0 24.91-9.87 24.91-22a19.85 19.85 0 00-5.5-13.83c5.54 4 9.26 10.35 9.26 17.19 0 12.17-11.16 22-24.91 22a29.63 29.63 0 01-20.34-8.96z" fill="url(#g)"/><ellipse cx="219.53" cy="161.32" rx="24.91" ry="22.04" fill="#fff73c"/><path d="M193.84 107.44a26.37 26.37 0 0016.57 5.56c13.76 0 24.91-9.87 24.91-22a19.82 19.82 0 00-5.5-13.83c5.54 4 9.26 10.35 9.26 17.2 0 12.17-11.15 22-24.91 22a29.61 29.61 0 01-20.33-8.93z" fill="url(#h)"/><ellipse cx="210.41" cy="90.99" rx="24.91" ry="22.04" fill="#ff2e5c"/><g><path d="M218.26 242.92l-9.09 4.36c-5.13 2.46-13.77-4.78-19.2-16.09l-49.22-112.38c-1.83-3.82-1.91-7.61-.18-8.44L154.8 103c1.73-.83 4.64 1.61 6.47 5.42l56.45 109.43c5.43 11.34 5.67 22.61.54 25.07z" fill="url(#i)"/><path d="M153.88 112.69l-6 2.88a6.87 6.87 0 01-9.14-3.21L125.31 78a2.31 2.31 0 011.09-3.08l9.49-4.71a2.32 2.32 0 013.11 1.03l18.12 32.31a6.88 6.88 0 01-3.24 9.14z" fill="#d9d9d9"/><path d="M87.73.21s16.75 14.14 4.47 51.35C87.36 67.19 110.06 88 130.52 75.37s12.28-36.83 4.84-46.14S98.15-2.39 87.73.21z" fill="url(#j)"/><path d="M100 16c1.12 5.59-3.6 32.09-3.16 40 .37 6.7 6.7 17.49 18.23 18-5.57-5.51-11.34-12-12.64-17.79S105 22.16 100 16z" fill="url(#k)"/><path d="M130 68.67s10.61-3.34 7.82-15.81-6.33-21-13.21-28.46-16.41-13.21-20.13-15.82c10.42 9.68 21.76 22.51 24.37 28.09s10.79 28.33 1.15 32z" fill="url(#l)"/><path d="M122.06 71.33s6.42-3.63 3.35-15.35-6.28-21.35-9.91-24.42c5 2.93 15.35 20 14.51 25.53s-2.65 13.96-7.95 14.24z" fill="url(#m)"/><path d="M111.31 60.44c-3-2.24-5.44-2.09-4-15.35s2.93-13.39 1.25-18.69S99.87 8.81 95.69 5.88C102 9.51 114.1 25 113.41 32s-1.26 18.69.28 21.21 2.65 6.69 1.81 9.62a8.67 8.67 0 00-.42 4.33c0 1.16-2.08-5.45-3.77-6.72z" fill="url(#n)"/></g></g></g></svg>