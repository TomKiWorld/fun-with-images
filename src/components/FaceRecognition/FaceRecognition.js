import React from 'react';
import ColorList from '../ColorList/ColorList';
import './FaceRecognition.css';

/**
 * Displays 
 * - Preview of submitted image url 
 * - Squares where faces were found
 * - Option to see colors found in the image
 * 
 * Required props:
 * - imageUrl => From state
 * - onShowClick => Function to change state of toggle color list
 */
const FaceRecognition = ({ imageUrl, colors }) => {
const imgBlock = imageUrl ? <img id='imageElement' src={imageUrl} alt='result' /> : <p className='white'>Submit an image url to start</p>
const colorsView = imageUrl ? 
  <ColorList 
    colors={colors} /> : 
    '';

  return (
    <section className='face-recognition container'>
      <div className='image-holder'>
        {imgBlock}
      </div>
        {colorsView}
    </section>
  );
}

export default FaceRecognition;
