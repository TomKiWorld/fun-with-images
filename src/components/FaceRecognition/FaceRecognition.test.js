import { shallow } from 'enzyme';
import React from 'react';
import FaceRecognition from './FaceRecognition';

describe('FaceRecognition Component', () => {
  const mockProps = {
    showColor: false
  }
  const imageUrl = 'https://www.sciencenewsforstudents.org/wp-content/uploads/2019/11/860_main_beauty.png'
  const mockShow = () => {
    return mockProps.showColor = !mockProps.showColor;
  }
  const wrapper =shallow(
    <FaceRecognition 
      imageUrl={imageUrl} 
      onShowClick={mockShow}/>
    );
  const colorBtn = wrapper.find('.view-colors').at(0);

  it('Expect to render ColorList Component', () => {     
    expect().toMatchSnapshot(wrapper);
  });

  it('Expect to change showColor', () => {    
    colorBtn.simulate('click');
    expect(mockProps.showColor).toBe(true);
    colorBtn.simulate('click');
    expect(mockProps.showColor).toBe(false);
  });

  it('Expect to close showColor', () => {
    const mockOpenState = {
      showColor: true
    };
    const mockOpenShow = () => {
      return mockOpenState.showColor = !mockOpenState.showColor;
    };
    const wrapper =shallow(
      <FaceRecognition 
        imageUrl={imageUrl} 
        onShowClick={mockOpenShow}/>
      );
    const colorBtn = wrapper.find('.view-colors').at(0);
    colorBtn.simulate('click');
    expect(mockOpenState.showColor).toBe(false);
    colorBtn.simulate('click');
    expect(mockOpenState.showColor).toBe(true);
  });

  it('Expect not to show image', () => {
    expect(shallow(<FaceRecognition 
      imageUrl={''} 
      onShowClick={mockShow}/>
    ).exists('img')).toBe(false);
  });
});
