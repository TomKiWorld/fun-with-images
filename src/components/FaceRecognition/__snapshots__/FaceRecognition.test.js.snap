// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceRecognition Component Expect to render ColorList Component 1`] = `Object {}`;
